src/opnsense/mvc/app/models/OPNsense/Base/Constraints/BaseConstraint.php

```diff
@@ -43,7 +43,7 @@ public function isEmpty($node)
         $node_class = get_class($node);
         if ($node_class == "OPNsense\Base\FieldTypes\BooleanField") {
             return empty((string)$node);
-        } elseif (empty((string)$node) || (string)$node == "0") {
+        } elseif (empty((string)$node->getCurrentValue()) || (string)$node->getCurrentValue() == "0") {
             return true;
         }
         return false;
```

Comment:
I BTW would be tempted to move this function to `BaseField`, WDYT?

Function usage:
![image](https://github.com/user-attachments/assets/a235599b-a5be-4330-b7fa-3b43fe449b89)

Reply:
I agree, but that means that `BooleanField` has a different implementation, which might even be cleaner anyway.

Reply:
done